internal/ocm/resource/cluster.go

```diff
@@ -44,6 +45,20 @@ func (c *Cluster) CreateNodes(clusterTopology rosaTypes.ClusterTopology, autoSca
 	}
 
 	if workerDiskSize != nil {
+		if clusterTopology == rosaTypes.Hcp {
+			err := diskValidator.ValidateNodePoolRootDiskSize(int(*workerDiskSize))
+			if err != nil {
+				return err
+			}
+		} else if clusterTopology == rosaTypes.Classic {
```

Comment:
is there the 3rd possible value for clusterTopology?
If not, don't need to check `if clusterTopology == rosaTypes.Classic`
if yes, needs to add an additional `else` block to handle it, maybe throw an error such as invalid topology

Reply:
no, it's just classic and HCP - I just thought it made sense to structure it like this in case we end up adding another one in the future. I can change it to be a regular if/else though